### PR TITLE
xorg pkgs: update versions and homepage cgit -> gitlab

### PR DESCRIPTION
**Pull request by wdconinc**
This PR updates the versions in the first commit, and updates additional homepages from the cgit mirror to the primary gitlab service now in use.

### CODE REVIEW

File: var/spack/repos/builtin/packages/libx11/package.py [hunk 1]
```diff
@@ -11,11 +11,14 @@ class Libx11(AutotoolsPackage, XorgPackage):
 
     homepage = "https://www.x.org/"
     xorg_mirror_path = "lib/libX11-1.6.7.tar.gz"
+    git = "https://gitlab.freedesktop.org/xorg/lib/libx11.git"
```
Comment (tldahlgren) [inline var/spack/repos/builtin/packages/libx11/package.py:14]: Curious.  Do you plan to add the ability to build a branch?

File: var/spack/repos/builtin/packages/libxaw3d/package.py [hunk 2]
```diff
@@ -25,6 +27,7 @@ class Libxaw3d(AutotoolsPackage, XorgPackage):
     depends_on("libxmu")
     depends_on("libxext")
     depends_on("libxpm")
+    depends_on("xproto@7.0.22:")
```
Comment (tldahlgren) [inline var/spack/repos/builtin/packages/libxaw3d/package.py:30]: (Asking for confirmation) Is this dependency new for the new versions?  Or does it apply to all listed versions?

File: var/spack/repos/builtin/packages/libxmu/package.py [hunk 3]
```diff
@@ -12,17 +12,20 @@ class Libxmu(AutotoolsPackage, XorgPackage):
...
 
     maintainers("wdconinc")
 
+    version("1.2.1", sha256="bf0902583dd1123856c11e0a5085bd3c6e9886fbbd44954464975fd7d52eb599")
+    version("1.2.0", sha256="b4686c4b4570044bcfc35bfaa3edbe68185ddf8e3250387f74a140c8e45afb2f")
     version("1.1.4", sha256="3091d711cdc1d8ea0f545a13b90d1464c3c3ab64778fd121f0d789b277a80289")
     version("1.1.2", sha256="e5fd4bacef068f9509b8226017205040e38d3fba8d2de55037200e7176c13dba")
 
     depends_on("libxt")
+    depends_on("libxt@1.1:", when="@1.2.0:")
```
Comment (tldahlgren) [inline var/spack/repos/builtin/packages/libxmu/package.py]: You don't need the `.0` patch version here.

File: var/spack/repos/builtin/packages/xev/package.py [hunk 4]
```diff
@@ -15,7 +15,7 @@ class Xev(AutotoolsPackage, XorgPackage):
     debugging and development tool, and should not be needed in normal
     usage."""
 
-    homepage = "https://cgit.freedesktop.org/xorg/app/xev"
+    homepage = "https://gitalb.freedesktop.org/xorg/app/xev"
```
Comment (tldahlgren) [inline var/spack/repos/builtin/packages/xev/package.py]:
```suggestion
    homepage = "https://gitlab.freedesktop.org/xorg/app/xev"
```

File: var/spack/repos/builtin/packages/libxxf86misc/package.py [hunk 5]
```diff
@@ -9,7 +9,7 @@
 class Libxxf86misc(AutotoolsPackage, XorgPackage):
     """libXxf86misc - Extension library for the XFree86-Misc X extension."""
 
-    homepage = "https://cgit.freedesktop.org/xorg/lib/libXxf86misc"
+    homepage = "https://gilab.freedesktop.org/xorg/lib/libXxf86misc"
```
Comment (tldahlgren) [inline var/spack/repos/builtin/packages/libxxf86misc/package.py]:
```suggestion
    homepage = "https://gitlab.freedesktop.org/xorg/lib/libXxf86misc"
```

**Review by tldahlgren — CHANGES_REQUESTED** [review]
Confirmed the new sha256s and all but 2 homepages.  The unconfirmed homepages have typos.  Also have a couple of questions.

File: var/spack/repos/builtin/packages/libx11/package.py (hunk 1 above)
Comment (wdconinc) [reply var/spack/repos/builtin/packages/libx11/package.py:14]: No plans to add branches, but this was a way to add the gitlab repository there for easy copy & paste by whoever adds future new versions, so they can run compares of the configure.ac files. I didn't want to change the official homepage to the gitlab repository through.

File: var/spack/repos/builtin/packages/libxaw3d/package.py (hunk 2 above)
Comment (wdconinc) [reply var/spack/repos/builtin/packages/libxaw3d/package.py:30]: This is based on commit https://gitlab.freedesktop.org/xorg/lib/libxaw3d/-/commit/34b1fb5d114b6c763c06dd9618c3d8ca11662e34, and from my reading I think it should have applied all along and only became an error due to compilers changing warnings to errors. Upon close reading now, it's possible that this commit needs to be a patch for older versions in order to compile with newer clang.

File: var/spack/repos/builtin/packages/libxmu/package.py (hunk 3 above)
Comment (wdconinc) [reply var/spack/repos/builtin/packages/libxmu/package.py]:
```suggestion
    depends_on("libxt@1.1:", when="@1.2:")
```

**Comment by alecbcs** [comment]
@tldahlgren does this PR pass your checks for homepages and versions now?

**Comment by tldahlgren** [comment]
> @tldahlgren does this PR pass your checks for homepages and versions now?

Thanks for the ping @alecbcs

**Review by tldahlgren — APPROVED** [review]
[Re]confirmed *all* homepages and new version sha256s.

**Pull request merged**